lib/Graph/Grad.cpp

```diff
@@ -267,8 +267,12 @@ Function *glow::differentiate(Function *F, const TrainingConfig &conf,
       // Gradient for BatchedReduceAddNode is TileNode,
       // repeating OutputG batch times.
       auto axis = BRA->getAxis();
-      auto Num = Input.dims()[axis] / OutputG.dims()[axis];
-      auto *TN = new TileNode("tile.grad", Input.getType(), OutputG, Num, axis);
+      auto Num = Input.dims()[axis];
+      std::vector<size_t> Dims{1};
+      Dims.insert(Dims.end(), OutputG.dims().begin(), OutputG.dims().end());
```

Comment:
This will only work if `axis = 0`. You need to do something like this:
```
std::vector<size_t> Dims(Input.dims());   // might not work, you may need to use Input.dims().vec()
Dims[axis] = 1;
```
This will create a vector with the dimensions of `Input` but with 1 on the reduced dimension.